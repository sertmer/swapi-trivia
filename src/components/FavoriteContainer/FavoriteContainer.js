import React from 'react';
import './FavoriteContainer.scss';
import CharacterCard from '../CharacterCard/CharacterCard';
import PropTypes from 'prop-types'

const FavoriteContainer = ({ characters }) => {
  let characterCards = characters.map(character => {
    return (
        <CharacterCard  
        className='character_card'
        key={character.name}
        name={character.name}
        homeworld={character.homeworld}
        population={character.population}
        species={character.species}
        films={character.films}
        isFavorite={character.isFavorite}
        unFavoriteThis={character.unFavoriteThis}
        />
        )
      })
      return (
        <main>
          {characterCards}
        </main>
      )
    }


export default FavoriteContainer

FavoriteContainer.propTypes = {
  characters: PropTypes.array.isRequired
}