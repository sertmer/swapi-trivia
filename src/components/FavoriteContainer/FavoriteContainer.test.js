import React from 'react'
import { shallow } from 'enzyme'
import FavoriteContainer from './FavoriteContainer'

describe('FavoriteContainer', () => {
  let wrapper
  let mockCharacters = [{
    name: 'bob',
    homeworld: ['home'],
    population: '10000',
    species: 'human',
    films: ['1', '2'],
    isFavorite: true,
    unFavoriteThis: jest.fn()
  }]

  beforeEach(() => {
    wrapper = shallow(<FavoriteContainer characters={mockCharacters} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})