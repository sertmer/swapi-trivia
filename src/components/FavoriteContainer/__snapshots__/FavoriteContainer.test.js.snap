// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteContainer should match the snapshot 1`] = `
<main>
  <CharacterCard
    className="character_card"
    films={
      Array [
        "1",
        "2",
      ]
    }
    homeworld="home"
    isFavorite={true}
    key="bob"
    name="bob"
    population="10000"
    species="human"
    unFavoriteThis={[MockFunction]}
  />
</main>
`;
